sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs

```diff
@@ -10,18 +10,36 @@
 
 namespace Azure.Identity
 {
+    /// <summary>
+    /// Enables authentication of a service principal in to Azure Active Directory using a X509 certificate that is assigned to it's App Registration. More information
+    /// on how to configure certificate authentication can be found here:
+    /// https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-certificate-credentials#register-your-certificate-with-azure-ad
+    /// </summary>
     public class ClientCertificateCredential : TokenCredential
```

Comment:
Not related to this change, but shouldn't this be `CertificateCredentialClient` according to [https://github.com/Azure/azure-sdk/blob/master/docs/design/dotnet/DesignGuidelines.mdk#L75](https://github.com/Azure/azure-sdk/blob/master/docs/design/dotnet/DesignGuidelines.mdk#L75) ? or something like that?

Reply:
In general yes, types which connect to the service are suffixed with "Client".  However, for authentication, we are using the convention that all classes used to store or obtain data needed for authentication are suffixed with "Credential".  I'm creating a PR to amend the guidelines to call this out.

Reply:
Never mind, just noticed that this is not the client itself

---
In reply to: [298278303](https://github.com/Azure/azure-sdk-for-net/pull/6731#discussion_r298278303) [](ancestors = 298278303)